integration-tests/cucumber-tests-platform/src/test/java/org/opensmartgridplatform/cucumber/platform/glue/steps/database/core/EventSteps.java

```diff
@@ -52,14 +52,22 @@ public void anEvent(final Map<String, String> data) {
         final Device device = this.deviceRepository
                 .findByDeviceIdentification(getString(data, PlatformKeys.KEY_DEVICE_IDENTIFICATION));
 
-        final Event event = new Event(device, getDateTime2(getString(data, PlatformKeys.DATE), DateTime.now()).toDate(),
+        final Event event = new Event(device, this.getDateTime(getString(data, PlatformKeys.DATE)).toDate(),
                 getEnum(data, PlatformKeys.EVENT_TYPE, EventType.class, EventType.DIAG_EVENTS_GENERAL),
                 getString(data, PlatformKeys.KEY_DESCRIPTION, ""),
                 getInteger(data, PlatformKeys.KEY_INDEX, PlatformDefaults.DEFAULT_INDEX));
 
         this.eventRepository.save(event);
     }
 
+    private DateTime getDateTime(final String dateTime) {
+        try {
+            return getDateTime2(dateTime, DateTime.now());
+        } catch (final IllegalArgumentException e) {
+            return DateTime.parse(dateTime);
+        }
+    }
+
```

Comment:
Extend the DateTimeHelper.getDateTime method with the "DateTime.parse(..)" part. It makes the utility method more generic and prevents adding date/time manipulations in the EventSteps class.

Note: the current version of DateTimeHelper.getDateTime(..) accepts Strings like "today at midnight + 5 hours". It can't handle absolute date/time values yet.

Reply:
It was clear to me that the current version accepts strings like that. Not very useful in the scenarios I created.

Reply:
Moved the parsing to the DateTimeHelper.